nim-mode.el

```diff
@@ -170,7 +170,7 @@
   (when (and (derived-mode-p 'org-mode)
              (fboundp 'org-in-src-block-p) (org-in-src-block-p))
     (modify-syntax-entry ?# ". 124b" nim-mode-syntax-table)
-    (modify-syntax-entry ?[ ". 23" nim-mode-syntax-table))
+    (modify-syntax-entry ?\[ ". 23" nim-mode-syntax-table))
```

Comment:
Ohh, I see! Another instance of me being confused as to why emacs is happy with a non escaped `[` there. I've got a lot to learn. :laughing:

Reply:
Yeah, without the escaped you will see

```
nim-mode.el:186:84: Error: Invalid read syntax: ")", 186, 84
```

See log in https://github.com/jcs-PR/nim-mode/actions/runs/6343881032/job/17232745497#step:5:119..

Reply:
Huh. What's going on with my emacs installation that it doesn't trigger that? :rofl: Thanks!

Reply:
Only Emacs lower than 28 is triggering this error, but I have no idea why. That's why it's good to have CI tests. ;)